shared/subprocess/proc.go

```diff
@@ -172,6 +156,13 @@ func (p *Process) start(fds []*os.File) error {
 	p.hasMonitor = true
 	go func() {
 		defer close(p.chExit)
+		if p.Stdout != nil && p.closeFds {
```

Comment:
I wondered why you've moved this bit into the go routine rather than where it was previously?